File: materials/5_js/tasks/normal/15.js
/**
 * Сжатие строки (Run Length Encoding)
 * Ввод: строка, например `aaabbbcc`
 * Вывод: сжатая форма, например `a3b3c2`
 */
const str = "wwwwaaadexxxxxx";
const arr = str.split("");
let newStr = '';
let tempLetter = arr[0];
let counter = 0;
newStr += arr[0];
let lastElement = arr.length;
for (letter of arr) {
  lastElement--;
  if (letter === tempLetter) {
    counter++;
    if (lastElement===0) {
      newStr += counter;
    }
  } else {
    newStr += counter;
    tempLetter = letter;
    counter = 1;
    newStr += letter;
    continue;
  }
}
alert(newStr); // w4a3d1e1x6

File: materials/6_python/tasks/normal/15.py
'''
Сжатие строки (Run Length Encoding)
Ввод: строка, например `aaabbbcc`
Вывод: сжатая форма, например `a3b3c2`
'''
s = input("Вве строку - ")
if not s:
  print('')
else:
  compressed = ''
  count = 1
  for i in range(1, len(s)):
    if s[i] == s[i-1]:
      count += 1
    else:
      compressed += s[i-1] + str(count)
      count = 1
  compressed += s[i-1] + str(count)
  print(compressed)
